apps/web/src/app/(signing)/sign/[token]/form.tsx

```diff
@@ -105,15 +110,62 @@ export const SigningForm = ({ document, recipient, fields }: SigningFormProps) =
                 Cancel
               </Button>
 
-              <Button
-                className="w-full"
-                type="submit"
-                size="lg"
-                disabled={!isComplete || isSubmitting}
+              <Dialog
+                open={showConfirmSignatureDialog}
+                onOpenChange={setShowConfirmSignatureDialog}
               >
-                {isSubmitting && <Loader className="mr-2 h-5 w-5 animate-spin" />}
-                Complete
-              </Button>
+                <DialogTrigger asChild>
+                  <Button
+                    className="w-full"
+                    type="button"
+                    size="lg"
+                    disabled={!isComplete || isSubmitting}
+                  >
+                    {isSubmitting && <Loader className="mr-2 h-5 w-5 animate-spin" />}
+                    Complete
+                  </Button>
+                </DialogTrigger>
+                <DialogContent>
+                  <div className="text-center">
+                    <div className="text-xl font-semibold text-neutral-800">Sign Document</div>
+                    <div className="text-muted-foreground mx-auto w-4/5 py-2 text-center">
+                      You are about to finish signing "{document.title}". Are you sure you?
+                    </div>
+                  </div>
+
+                  <DialogFooter>
+                    <div className="flex w-full flex-1 flex-nowrap gap-4">
+                      <Button
+                        type="button"
+                        className="dark:bg-muted dark:hover:bg-muted/80 flex-1  bg-black/5 hover:bg-black/10"
+                        variant="secondary"
+                        onClick={() => {
+                          setShowConfirmSignatureDialog(false);
+                        }}
+                      >
+                        Cancel
+                      </Button>
+
+                      <Button
+                        type="button"
+                        className="flex-1"
+                        disabled={!isComplete || isSubmitting}
+                        onClick={async () => {
```

Comment:
what do you think about extracting the code into a separate function?

Reply:
I'll improve it

Reply:
Can't wait 💪

Reply:
Can you take a look?